npm/index.ts

```diff
@@ -0,0 +1,196 @@
+
+import { PathReporter } from "io-ts/PathReporter";
+import download from "download";
+
+const Asset = t.type(
+  {
+    id: t.number,
+    name: t.string,
+  },
+  "Asset"
+);
+
+type Asset = t.TypeOf<typeof Asset>;
+
+const Release = t.type(
+  {
+    assets: t.array(Asset),
+  },
+  "Release"
+);
+
+type Release = t.TypeOf<typeof Release>;
+type OS = "macos" | "linux";
+
+// __dirname will point to the dist/ folder since the code is transpiled prior to being ran with node.
+const root = Path.resolve(__dirname, "..");
+const dist = Path.resolve(root, "dist");
+// Github api requires having a user agent header. This can be literally anything.
+const userAgent = "unsplash-intlc";
+
+const getErrorOrElse = (defaultErrorMessage: string) => (error: unknown) =>
+  error instanceof Error ? error : new Error(defaultErrorMessage);
+
+const execChildProcess = (command: string) =>
+  pipe(
+    TE.tryCatch(
+      () => pipe(Util.promisify(ChildProcess.exec), apply(command)),
+      getErrorOrElse(`Could not run ${command}`)
+    ),
+    TE.chain(({ stderr, stdout }) => {
+      if (stderr.length > 0) {
+        return TE.left(
+          new Error(`Error running command: ${stderr.toString()}`)
+        );
+      } else {
+        return TE.right(stdout);
+      }
+    })
+  );
+
+const decode =
+  <A>(codec: t.Decoder<unknown, A>) =>
+  (input: string): E.Either<Error, A> => {
+    return pipe(
+      E.tryCatch(
+        () => JSON.parse(input),
+        (_error) => new Error(`Failed at decoding ${codec.name}: ${input}`)
+      ),
+      E.chain((input) => {
+        const decoded = codec.decode(input);
+
+        // very strange stuff here...
+        if (E.isLeft(decoded)) {
+          return E.left(
+            pipe(
+              PathReporter.report(decoded),
+              (messages) => messages.join("\n"),
+              (message) => new Error(message)
+            )
+          );
+        } else {
+          return decoded;
+        }
+      })
+    );
+  };
+
+const getOSFromPlatform = (platform: NodeJS.Platform): O.Option<OS> => {
+  const isWindows = platform === "win32";
+  const isMacOs = platform === "darwin";
+
+  // We're kind of assuming here...
+  // https://gist.github.com/cvan/ef28f73b88b991d4a9705314b2af2e78
+  if (!isWindows && !isMacOs) {
+    return O.some("linux");
+  } else if (isMacOs) {
+    return O.some("macos");
+  } else {
+    return O.none;
+  }
+};
+
+const fetchAssetFromRelease = (tag: string, os: OS) =>
+  TE.tryCatch((): Promise<Asset> => {
+    return new Promise((resolve, reject) => {
+      let body: string = "";
+      const request = Https.get(
+        {
+          host: "api.github.com",
+          protocol: "https:",
+          path: `/repos/unsplash/intlc/releases/tags/${tag}`,
+          headers: {
+            "user-agent": userAgent,
+          },
+        },
+        (response) => {
+          response.setEncoding("utf8");
+
+          // Urgh...
+          response.on("data", (chunk) => {
```

Comment:
You could use `node-fetch` and then `TE.taskify` so you don't have to deal with all of this stuff…

Reply:
I wasn't sure how much dependencies we wanted to add, especially if we wanted to open source. Granted it has a bunch of fp-ts stuff at the moment.